### DISCUSSION TITLE
Export from single instance collection

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#10568] [Feature Requests]
Currently it is not possible to export data from single instance collections.
This is bad, as it makes the export feature useless if your website makes heavy use of single instance collections, which is the case for me.

Usually when I create a custom page in my website, I use the pattern of creating a single instance collection with some static data and then having a O2M relation for some dynamic data etc.

Now I want to export all the static data and also the dynamic data. Nothing of this is possible right now.

This is related to #6753, but not a duplicate, as this issue is more about not being able to export anything on single instance collections at all.

**Comment by rijkvanzanten** [comment]
For what it's worth, you can still export the single item collections directly through the API, by appending `?export=json` (or csv/xml) to the endpoint you wanna save the data of.

**Reply by LasseRosenow** [reply]
Thats a nice solution for developers, but our customers require it to be possible through the UI though.
Also we cannot send links to our customers to use, as its also not a deep export, so it only contains IDs in the O2M relation.
Right now the only real way to export it the way our customer wants, is through graphql queries, that also contain the nested data, but thats way to complicated for them 😢

**Reply by rijkvanzanten** [reply]
Yeah totally! Not saying it's the done-and-over with alternative. Just pointing out that there _is_ an alternative available 🙂 

To make that endpoint a accessible to your end users, you could setup a button-links presentation interface at the top of your page, that links to the endpoint with the query params pre-configured. That allows them to have a single "Download" button for the current item. As for the nested IDs, you can add any of the other regular query parameters (including the `fields` parameter) to fine-tune what you're exporting exactly 👍🏻

**Reply by LasseRosenow** [reply]
Wow thanks a lot, I did not know about the `fields` parameter, thats amazing :)
That would work for us for now, but in the long term having it in the gui would be ideal of course :)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
